packages/data-context/test/unit/sources/ProjectDataSource.spec.ts

```diff
@@ -143,46 +141,95 @@ describe('findSpecs', () => {
     'e2e/onboarding.spec.ts',
     'e2e/onboarding.cy.ts',
     'e2e/onboarding.cy.js',
+    'e2e/onboarding.cy.js.mp4',
   ]
 
   let ctx: DataContext
 
   beforeEach(async () => {
     ctx = createTestDataContext('run')
-    await ctx.fs.ensureDir(temporary)
-    await Promise.all(fixture.map((element) => ctx.fs.outputFile(path.join(temporary, element), '')))
+    await ctx.fs.ensureDir(projectRoot)
+    await Promise.all(fixture.map((element) => ctx.fs.outputFile(path.join(projectRoot, element), '')))
   })
 
   afterEach(async () => {
-    await ctx.fs.remove(temporary)
+    await ctx.fs.remove(projectRoot)
+  })
+
+  it('exludes specs outside `specPattern`, even if passing a generic glob', async () => {
```

Comment:
Would adding an e2e test covering this behavior be worth it? I feel like this behavior is pretty important and having more coverage around it would help since we don't have a test that covers the whole flow (starting at command line `--spec <specPattern>`).

Reply:
Sure, we can probably get it "for free" from a system test using `spec: '**/*` or something like this. I'll add it.

Reply:
Ok, I added it as an integration test in `cypress_spec.js`. https://github.com/cypress-io/cypress/pull/21462/commits/0b55b3146d2bde62c78ca0813ad0deb9d52893fb#diff-0416ec0277f3a43b360ddb7f3b56280f1ab1851592bc36e90fcffe74ee571ee8R396-R399

Notice I added a `qux.mp4` but we only call open browser once - that's because we detect `qux.cy.js`, but not the invalid `qux.mp4`.